netcfgbu/cli/probe.py

```diff
@@ -1,82 +1,46 @@
-import asyncio
+"""This module provides probing functionality to check the status of network devices."""
 
 import click
 
-from netcfgbu.logger import get_logger, stop_aiologging
-from netcfgbu.aiofut import as_completed
-from netcfgbu.probe import probe
-
-from .root import (
-    cli,
-    WithInventoryCommand,
-    opt_config_file,
-    opts_inventory,
-    opt_timeout,
-)
-
-from .report import Report
+from netcfgbu.cli.common import execute_command
+from netcfgbu.config_model import AppConfig
 from netcfgbu.consts import DEFAULT_PROBE_TIMEOUT
+from netcfgbu.probe import probe
 
+from .root import WithInventoryCommand, cli, opt_config_file, opt_timeout, opts_inventory
 
-def exec_probe(inventory, timeout=None):
-    inv_n = len(inventory)
-    log = get_logger()
-    log.info(f"Checking SSH reachability on {inv_n} devices ...")
-    timeout = timeout or DEFAULT_PROBE_TIMEOUT
-
-    loop = asyncio.get_event_loop()
-
-    tasks = {
-        probe(
-            rec.get("ipaddr") or rec.get("host"), timeout=timeout, raise_exc=True
-        ): rec
-        for rec in inventory
-    }
-
-    total = len(tasks)
-    done = 0
-    report = Report()
+CLI_COMMAND = "probe"
 
-    async def proces_check():
-        nonlocal done
 
-        async for probe_task in as_completed(tasks):
-            done += 1
-            task_coro = probe_task.get_coro()
-            rec = tasks[task_coro]
-            msg = f"DONE ({done}/{total}): {rec['host']} "
+def exec_probe(inventory_recs: list, timeout=None) -> None:
+    """Executes the probe command on the provided inventory records.
 
-            try:
-                probe_ok = probe_task.result()
-                report.task_results[probe_ok].append((rec, probe_ok))
+    Args:
+        inventory_recs: List of inventory records to probe.
+        timeout: Optional timeout value for the probe command. Defaults to DEFAULT_PROBE_TIMEOUT.
+    """
+    timeout = timeout or DEFAULT_PROBE_TIMEOUT
 
-            except (asyncio.TimeoutError, OSError) as exc:
-                probe_ok = False
-                report.task_results[False].append((rec, exc))
+    def task_creator(rec: dict, app_cfg: AppConfig):
+        """Creates a probe task for the given inventory record.
 
-            except Exception as exc:
-                probe_ok = False
-                log.error(msg + f"FAILURE: {str(exc)}")
+        Args:
+            rec: A dictionary representing an inventory record.
+            app_cfg: Application configuration object.
 
-            log.info(msg + ("PASS" if probe_ok else "FAIL"))
+        Returns:
+            A probe task configured with the IP address or hostname from the inventory record.
+        """
+        return probe(rec.get("ipaddr") or rec.get("host"), timeout=timeout, raise_exc=True)
```

Comment:
**Suggestion:** Consider handling possible exceptions when retrieving the IP address or hostname from the inventory record. This will prevent the probe task from failing silently if the record is missing these critical fields. [possible bug, importance: 9]
```suggestion
ip_or_host = rec.get("ipaddr") or rec.get("host")
if not ip_or_host:
    raise ValueError("Inventory record is missing 'ipaddr' or 'host'.")
return probe(ip_or_host, timeout=timeout, raise_exc=True)
```